shopfloor/services/zone_picking.py

```diff
@@ -510,17 +523,25 @@ def _scan_source_location(
             message = self.msg_store.location_not_allowed()
             return response, message
 
-        product, lot, package = self._find_product_in_location(location, product_id)
-        if len(product) > 1:
+        if package and package.location_id != location:
+            # Do not search based on a package from a previous location
+            package = False
+        product, lot, packages = self._find_product_in_location(
+            location, product_id, package
+        )
+        if len(packages) > 1 or (
+            len(packages) >= 1 and self.work.menu.scan_location_or_pack_first
+        ):
+            message = self.msg_store.several_packs_in_location(location)
```

Comment:
When `len(packages) == 1`, you cannot say that there are several packs in location
:astonished:

Reply:
Fixed